src/main/java/org/folio/circulation/domain/notice/schedule/ScheduledDigitalReminderHandler.java

```diff
@@ -206,22 +240,41 @@ private CompletableFuture<Result<ScheduledNoticeContext>> createFeeFineAction(Sc
    */
   @Override
   protected CompletableFuture<Result<ScheduledNotice>> updateNotice(ScheduledNoticeContext context) {
-    OverdueFinePolicyRemindersPolicy.ReminderSequenceEntry nextReminder = context.getLoan().getNextReminder();
+    RemindersPolicy.ReminderConfig nextReminder = context.getLoan().getNextReminder();
     if (nextReminder == null) {
       return deleteNotice(context.getNotice(), "no more reminders scheduled");
     } else if (isNoticeIrrelevant(context)) {
       return deleteNotice(context.getNotice(), "further reminder notices became irrelevant");
     } else {
-      ScheduledNotice nextReminderNotice = context.getNotice()
-        .withNextRunTime(nextReminder.getPeriod().plusDate(systemTime));
-      nextReminderNotice.getConfiguration()
-        .setTemplateId(nextReminder.getNoticeTemplateId())
-        .setFormat(nextReminder.getNoticeFormat());
-
-      return scheduledNoticesRepository.update(nextReminderNotice);
+      return findNextRuntimeAndBuildNotice(context, nextReminder)
+        .thenCompose(scheduledNoticeResult -> scheduledNoticesRepository.update(scheduledNoticeResult.value()));
     }
   }
 
+  protected CompletableFuture<Result<ScheduledNotice>> findNextRuntimeAndBuildNotice(
+    ScheduledNoticeContext context, RemindersPolicy.ReminderConfig nextReminder) {
+    log.debug("buildNextNotice:: parameters notice context: {}, reminder config: {}",
+      () -> context, () -> nextReminder);
+
+    return configurationRepository.findTimeZoneConfiguration()
+      .thenCompose(tenantTimeZone -> nextReminder.nextNoticeDueOn(systemTime,
+            tenantTimeZone.value(), context.getLoan().getCheckoutServicePointId(),
+            calendarRepository))
+        .thenApply(r -> r.next(nextRunTime -> succeeded(buildNextNotice(context, nextReminder, nextRunTime))
```

Comment:
Something is off with the formatting here.. More than two spaces indentation + .thenApply is not on the same level as .thenCompose